lib/steep/type_construction.rb

```diff
@@ -697,6 +697,36 @@ def synthesize(node)
 
           typing.add_typing(node, union_type(*types))
 
+        when :rescue
+          yield_self do
+            body, *resbodies, else_node = node.children
+            body_type = synthesize(body) if body
+            resbody_types = resbodies.map { |resbody| synthesize(resbody) }
+            else_type = synthesize(else_node) if else_node
```

Comment:
When `begin` has an `else`, `begin` body value is always ignored.
For example: `begin 1 else 2 end == 2 # => true`